File: tests/algorithms/x/PHP/data_structures/binary_tree/binary_tree_mirror.php
<?php
ini_set('memory_limit', '-1');
function _str($x) {
    if (is_array($x)) {
        $isList = array_keys($x) === range(0, count($x) - 1);
        if ($isList) {
            $parts = [];
            foreach ($x as $v) { $parts[] = _str($v); }
            return '[' . implode(' ', $parts) . ']';
        }
        $parts = [];
        foreach ($x as $k => $v) { $parts[] = _str($k) . ':' . _str($v); }
        return 'map[' . implode(' ', $parts) . ']';
    }
    if (is_bool($x)) return $x ? 'true' : 'false';
    if ($x === null) return 'null';
    return strval($x);
}
function binary_tree_mirror_dict(&$tree, $root) {
  if (($root == 0) || (!(array_key_exists($root, $tree)))) {
  return;
}
  $children = $tree[$root];
  $left = $children[0];
  $right = $children[1];
  $tree[$root] = [$right, $left];
  binary_tree_mirror_dict($tree, $left);
  binary_tree_mirror_dict($tree, $right);
}
function binary_tree_mirror($binary_tree, $root) {
  if (count($binary_tree) == 0) {
  $panic('binary tree cannot be empty');
}
  if (!(array_key_exists($root, $binary_tree))) {
  $panic('root ' . _str($root) . ' is not present in the binary_tree');
}
  $tree_copy = [];
  foreach (array_keys($binary_tree) as $k) {
  $tree_copy[$k] = $binary_tree[$k];
};
  binary_tree_mirror_dict($tree_copy, $root);
  return $tree_copy;
}
function main() {
  $binary_tree = [1 => [2, 3], 2 => [4, 5], 3 => [6, 7], 7 => [8, 9]];
  echo rtrim('Binary tree: ' . _str($binary_tree)), PHP_EOL;
  $mirrored = binary_tree_mirror($binary_tree, 1);
  echo rtrim('Binary tree mirror: ' . _str($mirrored)), PHP_EOL;
}
main();

File: tests/algorithms/x/Kotlin/data_structures/binary_tree/binary_tree_mirror.kt
fun panic(msg: String): Nothing { throw RuntimeException(msg) }

fun binary_tree_mirror_dict(tree: MutableMap<Int, MutableList<Int>>, root: Int): Unit {
    if ((root == 0) || (!(root in tree) as Boolean)) {
        return
    }
    var children: MutableList<Int> = (tree)[root] as MutableList<Int>
    var left: Int = (children[0]!!).toInt()
    var right: Int = (children[1]!!).toInt()
    (tree)[root] = mutableListOf(right, left)
    binary_tree_mirror_dict(tree, left)
    binary_tree_mirror_dict(tree, right)
}

fun binary_tree_mirror(binary_tree: MutableMap<Int, MutableList<Int>>, root: Int): MutableMap<Int, MutableList<Int>> {
    if (binary_tree.size == 0) {
        panic("binary tree cannot be empty")
    }
    if (!(root in binary_tree)) {
        panic(("root " + root.toString()) + " is not present in the binary_tree")
    }
    var tree_copy: MutableMap<Int, MutableList<Int>> = mutableMapOf<Int, MutableList<Int>>()
    for (k in binary_tree.keys) {
        (tree_copy)[k] = (binary_tree)[k] as MutableList<Int>
    }
    binary_tree_mirror_dict(tree_copy, root)
    return tree_copy
}

fun user_main(): Unit {
    var binary_tree: MutableMap<Int, MutableList<Int>> = (mutableMapOf<Int, MutableList<Int>>(1 to (mutableListOf(2, 3)), 2 to (mutableListOf(4, 5)), 3 to (mutableListOf(6, 7)), 7 to (mutableListOf(8, 9))) as MutableMap<Int, MutableList<Int>>)
    println("Binary tree: " + binary_tree.toString())
    var mirrored: MutableMap<Int, MutableList<Int>> = binary_tree_mirror(binary_tree, 1)
    println("Binary tree mirror: " + mirrored.toString())
}

fun main() {
    user_main()
}
